### DISCUSSION TITLE
How do I make the button highlight go away?

### DISCUSSION THREAD

**Discussion by mozeryansky** [chakra-ui/chakra-ui#6019] [General] [answered]
When I press a button it shows a light blue highlight, but it does not go away. How can I make the button only temporarily show the highlight?

https://chakra-ui.com/docs/components/form/icon-button

<img width="63" alt="Screen Shot 2022-05-13 at 2 13 16 AM" src="https://user-images.githubusercontent.com/3287719/168230945-a09b9ca6-be6c-4287-b040-77c30274cb3e.png">

**Comment by aroyan** [comment]
Try adding pseudo selector _focus as props in `<IconButton />` component.

For expample like this :
```javascript
    <IconButton icon={<AddIcon />} _focus={{outline: 'none'}}/>
```
You can pass any CSS property on that pseudo selector, in this case I don't want to show outline while clicking the Button so I passed `"none"` as a value of `outline` property.

Or checkout [codesanbox](https://codesandbox.io/s/xenodochial-mcclintock-ijimi0?file=/src/App.js) that I've created.

References : 
- [Style Props - Pseudo](https://chakra-ui.com/docs/styled-system/features/style-props#pseudo)
- [CSS outline property](https://developer.mozilla.org/en-US/docs/Web/CSS/outline)

**Reply by mozeryansky** [reply]
Do you know why the default is to keep the outline? It seems the opposite of normal UI behavior, just wondering if there is a design decision doc or something.

**Reply by mozeryansky** [reply]
Based on your answer, I figured out that the style is actually `_focus: { boxShadow: "outline" }`: https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/button.ts#L13

Thank you for your sandbox! I found that if I set the _focus as `{{}}` it also works the same, as it removes that style. However, I don't want to have to do this for every button. I am able to accomplish this with themes, I will add my own answer.

**Reply by aroyan** [reply]
> 

I don't know for sure why, but it seems Bootstrap is using the same style like this. Maybe it's improving user experience or something.

**Reply by mozeryansky** [reply]
That helps a lot, I searched the reason for Bootstrap and found a lot of people mentioning it's for accessibility when using keyboard navigation: https://stackoverflow.com/questions/23333231/bootstrap-button-shows-blue-outline-when-clicked

Either way, I'm glad I can change it.

**Comment by mozeryansky** [comment, marked as answer]
Extending from @aroyan's answer, below I extend the theme to solve my issue. And based on other comment https://github.com/chakra-ui/chakra-ui/discussions/5538#discussioncomment-2149191, I am basing the change from the default style: https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/button.ts

```
import { extendTheme } from "@chakra-ui/react"

const theme = extendTheme({
  components: {
    Button: {
      baseStyle: {
        _focus: {
          boxShadow: "none"
        }
      }
    }
  }
});

...
<ChakraProvider theme={theme}>
      <App />
</ChakraProvider>
...
```

I was not able to remove the style using just `{}` since `extendTheme` merges, so I reset the style to `"none"`. @aroyan answer is correct for a single button, but I wanted to remove it for all buttons.

**Reply by aroyan** [reply]
That's great
